### PR TITLE
tests: Add versions.yaml in order to support nginx working image

### LINKED ISSUE #700: Add versions.yaml in order to support nginx working image
With versions.yaml we will able to specify which nginx version is working

### PR DESCRIPTION
**Pull request by GabyCT**
With versions.yaml we will be able to specify which nginx version is
working.

Fixes #700

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by grahamwhaley** [comment]
Indeed, this is good to have. I think having it in its own separate file in this repo makes sense. As to txt/yaml....
- I agree using yaml like we do for the runtime versions file makes a lot of sense
    - except it will add some complexity to the test code - we will need a `ver=$(yq version.yaml myimage)` type line
- if we stick with a txt file, then I think that should probably be a '.bash' file, seeing as we are sourcing it - it is shell syntax/grammar really.
    - but sourcing that will start to 'pollute' the tests with a lot of image vars, only one of which each test is really interested in.

I think I vote yaml therefore - the `yq` line does not look too bad (my line above is not correct btw, but you can find/copy similar lines that read the runtime version.yaml file from this codebase). I also like the global shell name space pollution reduction.

**Comment by jodh-intel** [comment]
Sounds good to me.

We could potentially move some of the contents of `versions.yaml` into this repo too?

See also: https://github.com/kata-containers/packaging/issues/165.

**Comment by GabyCT** [comment]
@chavafg changes applied

**Comment by GabyCT** [comment]
@grahamwhaley and @jodh-intel , I did these changes :)

File: versions.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+  nginx:
+    description: "Proxy server for HTTP, HTTPS< SMTP, POP3 and IMAP protocols"
```
Comment (jodh-intel) [inline versions.yaml]: It looks like there might be a typo here - should that be?:

> HTTPS, SMTP,

File: versions.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+  nginx:
```
Comment (jodh-intel) [inline versions.yaml]: It's difficult to know the exact structure, but I would be inclined to have a "higher level" key as this file might grow over time. Similar to https://github.com/kata-containers/runtime/blob/master/versions.yaml, maybe you could do something like:

```
dependencies:
  nginx:
     ...
```

Or if not `dependencies`, maybe `externals` or something. I'd also consider adding in `description` and `format` as we do for the runtime equivalent of this file.

**Comment by GabyCT** [comment]
@jodh-intel , yes sorry it was a typo, fixed

File: versions.yaml (hunk 2 above)
Comment (chavafg) [reply versions.yaml]: agree with @jodh-intel, maybe call them `docker_images`? we later can add all docker images that we use:

```
docker_images:
  description: "Docker images used for testing"

  nginx:
    ....
```

**Comment by GabyCT** [comment]
@chavafg and @jodh-intel changes applied

**Comment by jcvenegas** [comment]
@grahamwhaley  @jodh-intel  please  check again and merge if you are happy with the current changes.

**Pull request merged**